src/lint/collect-header-diagnostics.ts

```diff
@@ -46,7 +46,7 @@ export function collectHeaderDiagnostics(
       // Object.fromEntries
       // ForIn/OfHeadEvaluation
       // Array.prototype [ @@iterator ]
-      /^[A-Za-z][A-Za-z0-9/]*(\.[A-Za-z][A-Za-z0-9/]*)*( \[ @@[a-z][a-zA-Z]+ \])?\s*$/,
+      /^[A-Za-z][A-Za-z0-9/]*(\.[A-Za-z][A-Za-z0-9]*)*((\.__[a-z][A-Za-z0-9]*__)?|( \[ @@[a-z][a-zA-Z]+ \])?)\s*$/,
```

Comment:
```suggestion
      /^[A-Za-z][A-Za-z0-9/]*(\.[A-Za-z][A-Za-z0-9]*)*(\.__[a-z][A-Za-z0-9]*__| \[ @@[a-z][a-zA-Z]+ \])?\s*$/,
```

Don't put two optional things in a disjunction.

Reply:
Pulled the `?` to outside the disjunction.